### PR TITLE
refactor(Core): remove ace_autoptr, cleanup

### PR DESCRIPTION
**Pull request by Viste**
## CHANGES PROPOSED:
-  remove ace_auto_ptr switch to uniq, bit cleanup
-  @FrancescoBorzi as you asked

## HOW TO TEST THE CHANGES:
play on server understand that all is ok also try some modules

## Target branch(es):
- [x] Master
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Server/WorldSocket.cpp [hunk 1]
```diff
@@ -660,7 +660,7 @@ int WorldSocket::ProcessIncoming(WorldPacket* new_pct)
     ACE_ASSERT (new_pct);
 
     // manage memory ;)
-    ACE_Auto_Ptr<WorldPacket> aptr (new_pct);
+    std::unique_ptr<WorldPacket> aptr (new_pct);
```
Comment (Viste) [inline src/server/game/Server/WorldSocket.cpp:663]: i guess must be
`std::unique_ptr<WorldPacket> aptr(new WorldPacket(new_pct));`

**Comment by IntelligentQuantum** [comment]
@Viste 

`WorldSocket::handle_input_header(): client (account: 0, char [GUID: 0, name: <none>]) sent malformed packet (size: 281, cmd: 808714240)`

i can't login
As you said, worldsocket have a problem

**Comment by Viste** [comment]
@IntelligentQuantum try now

**Comment by IntelligentQuantum** [comment]
NOPE

`
WorldSocket::handle_input_header(): client (account: 0, char [GUID: 0, name: <none>]) sent malformed packet (size: 281, cmd: 808714240)`

**Comment by Viste** [comment]
@IntelligentQuantum i update this shit try now

**Comment by IntelligentQuantum** [comment]
Still.....

`WorldSocket::handle_input_header(): client (account: 0, char [GUID: 0, name: <none>]) sent malformed packet (size: 281, cmd: 808714240)
`
works fine with previous edit
https://paste.ubuntu.com/p/mbSZxJWVjR/

**Comment by Viste** [comment]
> Still.....
> 
> `WorldSocket::handle_input_header(): client (account: 0, char [GUID: 0, name: <none>]) sent malformed packet (size: 281, cmd: 808714240) `
> works fine with previous edit
> https://paste.ubuntu.com/p/mbSZxJWVjR/
according to your diff in paste you use not last version X_x

**Comment by IntelligentQuantum** [comment]
I tried this, it compiles but it has a login problem

https://imgur.com/nV87ypX

`WorldSocket::handle_input_header(): client (account: 0, char [GUID: 0, name: <none>]) sent malformed packet (size: 281, cmd: 808714240) `

https://github.com/Viste/az-core/tree/aptrclean

I also tested this, it can be compiled, the login also works properly

https://imgur.com/wUyMxJp
https://imgur.com/l1czkTm

https://github.com/WarheadCore/WarheadCore/commit/8a27bafc281f28f752d3529928ecb9fcb53d1f78

- Arch Linux
- Clang Version 10.0.1
- Ace Version 6.5.10-1
- OpenSSL 1.1.1g

**Comment by Viste** [comment]
rabase can close PR :DD 
@FrancescoBorzi  redy for test

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum @azerothcore/testers  can you please test this PR?

**Comment by Viste** [comment]
@FrancescoBorzi 
Снимок экрана от 2020-08-25 06-46-25

**Comment by IntelligentQuantum** [comment]
Tested

`
AzerothCore rev. 4309a65a828e 2020-08-25 11:12:29 +0300 (aptrclean branch) (Unix, Release)
Connected players: 1. Characters in world: 1.
Connection peak: 1.
Server uptime: 13 minute(s) 17 second(s).
Update time diff: 10ms, average: 13ms.
`

**Comment by FrancescoBorzi** [comment]
good job @Viste , if you have more PRs like this, please submit them. It's always nice to get rid of dependencies where they are not needed.

thanks @IntelligentQuantum for testing

**Comment by boom8866** [comment]
Error: the unfinished achievement cannot be posted in the chat box and sent out

The achieved achievement can be sent

**Comment by FrancescoBorzi** [comment]
@boom8866  ?

**Comment by boom8866** [comment]
Attempt to send in chat box, unfinished achievement   @FrancescoBorzi

**Comment by FrancescoBorzi** [comment]
@boom8866 I don't understand what you are saying but if you found a bug, better open a bug report. This is a PR, and has already been merged.

If you think this PR caused the bug, you can mention it in the bug report.

**Pull request merged**